src/drivers/bitbucket_cloud.js

```diff
@@ -0,0 +1,79 @@
+const fetch = require('node-fetch');
+const FormData = require('form-data');
+const { URL, URLSearchParams } = require('url');
+const {Base64} = require('js-base64');
+
+const { fetch_upload_data } = require('../utils');
+
+class BitBucketCloud {
+  constructor(opts = {}) {
+    const { repo, token } = opts;
+
+    if (!token) throw new Error('token not found');
+    if (!repo) throw new Error('repo not found');
+
+    this.token = token;
+    this.repo = repo;
+
+    const { protocol, host, pathname } = new URL(this.repo);
+    this.repo_origin = `${protocol}//${host}`;
+    this.api = 'https://api.bitbucket.org/2.0';
+    this.project_path = encodeURIComponent(pathname.substring(1));
+  }
+
+  async comment_create(opts = {}) {
+    const { project_path } = this;
+    const { commit_sha, report} = opts;
+
+    // Print some variables
+    const endpoint = `/repositories/${project_path}/commit/${commit_sha}/comments/`;
+    const body= JSON.stringify({"content": {"raw": report}});
+
+    const output = await this.request({ endpoint, method: 'POST', body });
+
+    return output;
+  }
+
+  async check_create() {
+    throw new Error('BitBucket Cloud does not support check!');
+  }
+
+  async upload(opts = {}) {
+    throw new Error('BitBucket Cloud does not support upload!');
+  }
+
+  async runner_token() {
+    throw new Error('BitBucket Cloud does not support runner_token!');
+  }
+
+  async register_runner(opts = {}) {
+    throw new Error('BitBucket does not support register_runner!');
+  }
+
+  async unregister_runner(opts = {}) {
+    throw new Error('BitBucket does not support unregister_runner!');
+  }
+
+  async runner_by_name(opts = {}) {
+    throw new Error('BitBucket does not support runner_by_name!');
+  }
+
+  async request(opts = {}) {
+    const { token, api } = this;
+    const { endpoint, method = 'GET', body } = opts;
+
+    if (!endpoint) throw new Error('BitBucket API endpoint not found');
+    const {BITBUCKET_WORKSPACE}=process.env;
+    const headers = {
+      'Content-Type': 'application/json',
+      'Authorization': 'Basic ' + Base64.encode(`${BITBUCKET_WORKSPACE}:${token}`)};
```

Comment:
Lets require the token full. I mean the user has to provide is the ```repo_token``` or ```BITBUCKET_TOKEN``` as env or as a parameter and thats the base64 of their user and password. They have to calculate manually but I think is fine. What do you think?

```js
const headers = {
      'Content-Type': 'application/json',
      'Authorization': `Basic ${token}`
};
```

I have thought also to extend the input with ```user``` ```pass``` alternatively to ```token``` so we can do it for the user but Im not sure yet. What do you think?

Reply:
so you are saying that the user will need to manually generate a token that is 

base64($username:$password)? 

The only concern i have is... will this force them to write a little base64 encoder, or use some online widget to do it? I don't like asking people to manually create the key. I imagine a lot of users getting lost at this step or quitting, unless they'res an easy way to do I don't know. 

Not sure I understand your idea about extending the input, maybe if you make some sample code I could understand better!

Reply:
@elleobrien 
instead of using token they would have to setup user and password like this

```sh
cml-send-comment --user $MYUSER --pass $MYPASS report.md
```

alternatively they could set 
```sh
cml-send-comment --token $MYTOKEN report.md
```

where token is the base64($username:$password)

Reply:
I'm trying to understand what is the fundamental issue here. Why BB cannot be implemented the same way as GH/GL with a single token? Will the problem be solved by the separation of BB-cloud & BB-server drivers? I'd appreciate it if you can explain it.

Passwords and secrets in the command line is a potential security breach. In some settings, these passwords will end up in a centralized logging system with wide access across org. It needs to be avoided as much as possible.